.github/workflows/ci.yml

```diff
@@ -5,134 +5,237 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
+env:
+  DEFAULT_NODE_VERSION: 22.x
+
 jobs:
 
+  precheck:
```

Comment:
Linter errors are caught at the beginning by the precheck, as well as dependency issues (via depcheck). These are both quick to fix and quick to detect, which is why I added the check at the beginning. It also simulates the local experience (and pre-commit hook) of running the linter first, before buildind and testing. The lint and depcheck tasks also do not need to be run as part of the later tasks, since they are run only once here.

The Windows job has been split into different parallel subtasks, as well, and the longest-running Windows job now takes 7m 32s (the spec+integration tests), as opposed to the previous combined job of 12m 16s. I took these numbers by comparing these two runs: [old CI](https://github.com/comunica/comunica/actions/runs/11164579820), [new CI](https://github.com/comunica/comunica/actions/runs/11121288043). This more than compensates for the pre-check task's 1-minute runtime, I think.

Reply:
The speedup of windows is definitely nice.
But that seems orthogonal to this issue.

You just made me realize there's a big problem with this approach.
Namely that linting and depcheck won't be invoked anymore on mac and windows.
This means that we won't know when these checks would break on these environments.

Reply:
I had the amazing idea to run the preliminary checks on all three operating systems, to make sure the scripts work *and* to make sure the Yarn dependency cache is created for all three of them. That way, all the subsequent parallel tasks could immediately use this cache, while still keeping the preliminary checks separate.

Unfortunately, the CI seems to have slowed down a lot for no reason, it even ran some identical tasks at half the speed. :snail: I will look into this later, maybe it will be faster when I run it again.